smallab/runner_implementations/abstract_runner.py

```diff
@@ -0,0 +1,34 @@
+import json
+import pickle
+
+import os
+
+import abc
+import typing
+
+from smallab.experiment import Experiment
+
+
+class AbstractRunner(abc.ABC):
```

Comment:
This class is hard to parse. Here's a couple suggestions to make it easier for users to understand these method headers:

1. Add docstrings to explain what each arguement represents in more detail
2. For composite typing specifications, especially those that get reused often, it might be better to alias their type specification, e.g.
```
# in file types.py
spec = typing.List[typing.Dict]

# in file abstract_runner.py
from smallab import types
...
    def run(self, specifications_to_run: types.spec, ...
```
It would be more appropriate to use `typing.NewType()` and `typing.TypeVar()`, but looking through the docs for that gives me a headache...

Reply:
I don't think we should have a composite type representing a list of specifications but i will make one for a specification.